extension/src/popup/views/GrantAccess.tsx

```diff
@@ -66,24 +70,20 @@ const ButtonContainerEl = styled.div`
 const RejectButtonEl = styled(Button)`
   background: ${COLOR_PALETTE.text};
 `;
-const WarningMessageEl = styled.div`
-  background-color: ${COLOR_PALETTE.errorFaded};
-  padding: 0.5rem 1rem;
-  text-align: center;
-  font-size: 0.85rem;
-`;
 
 export const GrantAccess = () => {
   const location = useLocation();
   const dispatch = useDispatch();
+  const [isGranting, setIsGranting] = useState(false);
+
   const { url } = parsedSearchParam(location.search);
   const hostname = getUrlHostname(url);
   const punycodedDomain = punycode.toASCII(hostname);
-  const [isGranting, setIsGranting] = useState(false);
-
   const whitelistStr = localStorage.getItem(WHITELIST_ID) || "";
   const whitelist = whitelistStr.split(",");
-  const isDomainWhiteListed = whitelist.includes(removeQueryParam(url));
+  const isDomainWhiteListed = whitelist.includes(
+    removeQueryParam(punycodedDomain),
+  );
```

Comment:
ah didn't realize that we added some whitelist handling in `popup`. 

We should actually be doing all of this in `background`. I set up a pattern to follow here: https://github.com/stellar/lyra/blob/master/extension/src/background/messageListener/lyraApiMessageListener.ts#L31

You don't need to do fix this in this PR since it's already in master. I'll make a followup ticket